### PR TITLE
docs: update Module Federation guide

### PR DESCRIPTION
**Pull request by chenjiahan**
## Summary

Update Module Federation guide:

- Add Module Federation v2
- Add some comparisons
- Add some basic examples
- No longer recommend to use Module Federation v1.0

## Checklist

- [ ] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: website/docs/zh/guide/features/module-federation.mdx [hunk 1]
```diff
@@ -4,20 +4,81 @@ import { ApiMeta, Stability } from '@components/ApiMeta';
...
+
+你可以通过 Rspack 的 [ModuleFederationPluginV1](/plugins/webpack/module-federation-plugin-v1) 来使用它。
+
+:::tip
+Module Federation v1.0 已经不再迭代，我们推荐使用 Module Federation v1.5 或 v2.0 版本。
+:::
+
+### Module Federation v1.5
+
+这是 Rspack 内置支持的版本，提供模块导出、模块加载、依赖共享等能力。
```
Comment (ahabhgk) [inline website/docs/zh/guide/features/module-federation.mdx:42]:
```suggestion
这是 Rspack 内置支持的版本，除了支持 Module Federation v1.0 的模块导出、模块加载、依赖共享等能力之外，还添加了运行时插件功能，允许用户扩展模块联邦的行为、功能。
```

File: website/docs/zh/guide/features/module-federation.mdx [hunk 2]
```diff
@@ -4,20 +4,81 @@ import { ApiMeta, Stability } from '@components/ApiMeta';
...
+    }),
+  ],
+};
+```
+
+> 参考：[Module Federation v1.5 示例](https://github.com/rspack-contrib/rspack-examples/tree/main/rspack/module-federation-v1.5)。
+
+### Module Federation v2.0
+
+这是 Module Federation 的增强版本。它在 Module Federation v1.5 的基础上，提供了额外的功能，比如动态类型提示、manifest、federation runtime、runtime plugin system 等，这些功能让 Module Federation 更适用于支持大型 Web 应用的微前端架构。
```
Comment (ahabhgk) [inline website/docs/zh/guide/features/module-federation.mdx:66]:
```suggestion
这是 Module Federation 的增强版本。它基于 Module Federation v1.5，并在其基础上，开箱即用的提供了一些额外的功能，比如动态类型提示、manifest 等，这些功能让 Module Federation 更适用于支持大型 Web 应用的微前端架构。
```

File: website/docs/zh/plugins/webpack/module-federation-plugin-v1.mdx [hunk 3]
```diff
@@ -1,13 +1,31 @@
...
+module.exports = {
+  plugins: [
+    new rspack.container.ModuleFederationPluginV1({
+      // options
+    }),
+  ],
+};
+```
+
+The options for ModuleFederationPluginV1 are basically the same as those for [ModuleFederationPlugin](/plugins/webpack/module-federation-plugin), but it does not support the `implementation` and `runtimePlugins` fields.
```
Comment (ahabhgk) [inline website/docs/zh/plugins/webpack/module-federation-plugin-v1.mdx:31]:
```suggestion
ModuleFederationPluginV1 的配置基本与 [ModuleFederationPlugin](https://github.com/plugins/webpack/module-federation-plugin) 相同，但不支持 `implementation` 和 `runtimePlugins` 字段。
```

File: website/docs/en/guide/features/module-federation.mdx [hunk 4]
```diff
@@ -4,19 +4,81 @@ import { ApiMeta, Stability } from '@components/ApiMeta';
...
+
+You can use it through Rspack's [ModuleFederationPluginV1](/plugins/webpack/module-federation-plugin-v1).
+
+:::tip
+Module Federation v1.0 is no longer being iterated. We recommend using Module Federation v1.5 or v2.0.
+:::
+
+### Module Federation v1.5
+
+This version is natively supported in Rspack, providing features such as module export, module loading, dependency sharing.
```
Comment (ahabhgk) [inline website/docs/en/guide/features/module-federation.mdx:42]:
```suggestion
This version is natively supported in Rspack. In addition to supporting module export, module loading, and dependency sharing capabilities of Module Federation v1.0, it also adds runtime plugin functionality, allowing users to extend the behavior and functionality of module federation.
```

File: website/docs/en/guide/features/module-federation.mdx [hunk 5]
```diff
@@ -4,19 +4,81 @@ import { ApiMeta, Stability } from '@components/ApiMeta';
...
+    }),
+  ],
+};
+```
+
+> Refer to: [Module Federation v1.5 Example](https://github.com/rspack-contrib/rspack-examples/tree/main/rspack/module-federation-v1.5).
+
+### Module Federation v2.0
+
+This is the enhanced version of Module Federation. It provides additional features based on Module Federation v1.5, such as dynamic type hinting, manifest, federation runtime, runtime plugin system, etc. These features make Module Federation more suitable for supporting micro-front-end architecture of large web applications.
```
Comment (ahabhgk) [inline website/docs/en/guide/features/module-federation.mdx:66]:
```suggestion
This is the enhanced version of Module Federation. Based on Module Federation v1.5, it provides some additional features out of the box, such as dynamic type hints and manifest, etc. These features make Module Federation more suitable for supporting micro-front-end architecture of large web applications.
```

**Comment by chenjiahan** [comment]
@ahabhgk Updated ❤️ Thanks for the review

**Review by ahabhgk — APPROVED** [review]

**Pull request merged**